Add test cases for SocketUrlDetection

fuji-151a opened the pull request:
There is no test class in SocketUrlDetection.
So, I created SocketUrlDetectionTest class and I wrote the test cases.
Please check this pull request.

Thanks.

erikdw commented on storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java:
```diff
@@ -0,0 +1,95 @@
+            e.printStackTrace();
+        }
+    }
+}
```
Please add a newline at the end of the file.  i.e., the final character in the file should be a newline at the end of the `}` line.  I really dislike this behavior of some IDEs and/or editors to not finish lines with a newline...

fuji-151a replied:
OK. I'll try to rewrite code to put a newline at the end of the file.

erikdw commented on storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java:
```diff
@@ -0,0 +1,95 @@
+    private SocketUrlDetection target;
+
+    /**
+     * ServerSocket.
```
This comment isn't providing much value, I'd just ditch it.

fuji-151a replied:
I see. Then I'll delete this comment.

erikdw commented on storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java:
```diff
@@ -0,0 +1,95 @@
+     * Close server socket connection.
+     */
+    @AfterClass
+    public static void afeterClass() {
```
typo:  `afeterClass` --> `afterClass`.

fuji-151a replied:
Thank you for review.
I'll modify this typo.

erikdw commented on storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java:
```diff
@@ -0,0 +1,95 @@
+     */
+    @Test
+    public void isReachableFalse() {
+        target = new SocketUrlDetection(9999);
```
I'd put the magic port number of `9999` into a class-level constant.

fuji-151a replied:
OK, I'll try to use constant to this code.

erikdw commented on storm/src/test/storm/mesos/logviewer/SocketUrlDetectionTest.java:
```diff
@@ -0,0 +1,94 @@
+     */
+    @Test
+    public void isReachableFalse() {
+        target = new SocketUrlDetection(PORT);
```
Can you add a comment about how this test is assuming that nothing is listening to the chosen port on the machine where the test is run?   i.e., it's totally possible that something could be listening to that port.

fuji-151a replied:
@erikdw OK. I'll try to comment to this test case. 
I have one consultation. I have already used `9999` port in this code. but I think that `9999` port is often used. So, Should I change this port from `9999` to any port which are not used much?

erikdw replied:
@fuji-151a : good question... you'd want it to *not* be an ephemeral port, so I would avoid 32768+.  Also, Mesos Slave/Agent advertises 31000-32000 by default. [1] Maybe we can go with 29999 or something similar?  And let's change the `PORT` constant name to something  like `HOPEFULLY_UNUSED_PORT`?

Reference:
[1] https://unix.stackexchange.com/questions/211647/how-safe-is-it-to-change-the-linux-ephemeral-port-range/237543#237543

fuji-151a replied:
@erikdw Thank you for your answering my questions.
OK, I'll rewrite port number from `9999` to `29999`.
And I'll rewrite constant name from `PORT` to `HOPEFULLY_UNUSED_PORT`

fuji-151a commented:
@erikdw Thank you for review.
I rewrote code based on your review.
- modified constant name: PORT --> HOPEFULLY_UNUSED_PORT
- chaged port 9999 --> 29999
- added comment to isReachableFalse method

Please check again.

erikdw commented:
@fuji-151a : thanks for the great comments that make it very clear what is happening.
